### PR TITLE
Improve Windows dependencies instructions

### LINKED ISSUE #43: Windows build question - small BUILDING.md improvement proposal
Hi there, so I am using Windows 11 and following instructions to build touchHLE on my end.

I have downloaded boost from their website, and extracted it to `vendor/boost` directory. So the file structure is: `vendor/boost/boost_1_81_0`.

That might be wrong because the build is failing with the following log:

```
PS D:\Users\APereira\src\touchHLE> cargo run
   Compiling sdl2 v0.35.1
   Compiling touchHLE_dynarmic_wrapper v0.1.0 (D:\Users\APereira\src\touchHLE\src\cpu\dynarmic_wrapper)
   Compiling sdl2-sys v0.35.1
   Compiling touchHLE_openal_soft_wrapper v0.1.0 (D:\Users\APereira\src\touchHLE\src\audio\openal_soft_wrapper)
   Compiling touchHLE v0.1.0 (D:\Users\APereira\src\touchHLE)
error: failed to run custom build command for `touchHLE_dynarmic_wrapper v0.1.0 (D:\Users\APereira\src\touchHLE\src\cpu\dynarmic_wrapper)`

Caused by:
  process didn't exit successfully: `D:\Users\APereira\src\touchHLE\target\debug\build\touchHLE_dynarmic_wrapper-6fd497a82964d59d\build-script-build` (exit code: 101)
  --- stdout
  CMAKE_TOOLCHAIN_FILE_x86_64-pc-windows-msvc = None
  CMAKE_TOOLCHAIN_FILE_x86_64_pc_windows_msvc = None
  HOST_CMAKE_TOOLCHAIN_FILE = None
  CMAKE_TOOLCHAIN_FILE = None
  CMAKE_GENERATOR_x86_64-pc-windows-msvc = None
  CMAKE_GENERATOR_x86_64_pc_windows_msvc = None
  HOST_CMAKE_GENERATOR = None
  CMAKE_GENERATOR = None
  CMAKE_PREFIX_PATH_x86_64-pc-windows-msvc = None
  CMAKE_PREFIX_PATH_x86_64_pc_windows_msvc = None
  HOST_CMAKE_PREFIX_PATH = None
  CMAKE_PREFIX_PATH = None
  CMAKE_x86_64-pc-windows-msvc = None
  CMAKE_x86_64_pc_windows_msvc = None
  HOST_CMAKE = None
  CMAKE = None
  running: "cmake" "D:\\Users\\APereira\\src\\touchHLE\\src\\cpu\\dynarmic_wrapper\\../../..\\vendor/dynarmic" "-G" "Visual Studio 17 2022" "-Thost=x64" "-Ax64" "-DDYNARMIC_WARNINGS_AS_ERRORS=OFF" "-DDYNARMIC_TESTS=OFF" "-DBoost_INCLUDE_DIR=D:\\Users\\APereira\\src\\touchHLE\\src\\cpu\\dynarmic_wrapper\\../../..\\vendor/boost" "-DCMAKE_INSTALL_PREFIX=D:\\Users\\APereira\\src\\touchHLE\\target\\debug\\build\\touchHLE_dynarmic_wrapper-4f0f57d37945b485\\out" "-DCMAKE_C_FLAGS= -nologo -MD -Brepro" "-DCMAKE_C_FLAGS_DEBUG= -nologo -MD -Brepro" "-DCMAKE_CXX_FLAGS= -nologo -MD -Brepro" "-DCMAKE_CXX_FLAGS_DEBUG= -nologo -MD -Brepro" "-DCMAKE_ASM_FLAGS= -nologo -MD -Brepro" "-DCMAKE_ASM_FLAGS_DEBUG= -nologo -MD -Brepro" "-DCMAKE_BUILD_TYPE=Debug"
  -- Target architecture: x86_64
  -- Configuring incomplete, errors occurred!
  See also "D:/Users/APereira/src/touchHLE/target/debug/build/touchHLE_dynarmic_wrapper-4f0f57d37945b485/out/build/CMakeFiles/CMakeOutput.log".
  See also "D:/Users/APereira/src/touchHLE/target/debug/build/touchHLE_dynarmic_wrapper-4f0f57d37945b485/out/build/CMakeFiles/CMakeError.log".

  --- stderr
  CMake Error at C:/Program Files/CMake/share/cmake-3.25/Modules/FindBoost.cmake:1826 (file):
    file STRINGS file
    "D:\Users\APereira\src\touchHLE\src\cpu\dynarmic_wrapper\../../..\vendor/boost/boost/version.hpp"
    cannot be read.
  Call Stack (most recent call first):
    CMakeLists.txt:127 (find_package)


  CMake Error at C:/Program Files/CMake/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:225 (set):
    Syntax error in cmake code at

      C:/Program Files/CMake/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:225

    when parsing string

      Could NOT find Boost: Found unsuitable version "0.0.0", but required is at least "1.57" (found D:\Users\APereira\src\touchHLE\src\cpu\dynarmic_wrapper\../../..\vendor/boost, )

    Invalid character escape '\U'.
  Call Stack (most recent call first):
    C:/Program Files/CMake/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:598 (_FPHSA_FAILURE_MESSAGE)
    C:/Program Files/CMake/share/cmake-3.25/Modules/FindBoost.cmake:2377 (find_package_handle_standard_args)
    CMakeLists.txt:127 (find_package)


  thread 'main' panicked at '
  command did not execute successfully, got: exit code: 1

  build script failed, must exit now', D:\Users\APereira\.cargo\registry\src\github.com-1ecc6299db9ec823\cmake-0.1.49\src\lib.rs:1104:5
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I see that it tries to read `vendor/boost/boost` and the `boost_1_81_0` has a `boost` directory inside, so maybe the instructions could be updated to "From the boost zip package, extract the contents of `boost_n_nn_n` directory to `vendor/boost` directory" to make it clearer. Can I propose a small PR to update it? :)

### PR DESCRIPTION
**Pull request by nighto**
Closes #43

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -2,11 +2,13 @@
 
 You need [git](https://git-scm.com/), [the Rust toolchain](https://www.rust-lang.org/tools/install), and your platform's standard C and C++ compilers.
 
+* On Windows, notice that you must run the Rust toolchain installer from a terminal instead of double clicking it.
```
Comment (hikari-no-yume) [inline BUILDING.md]: Are you sure it doesn't work otherwise? It's just an exe, it will open a command prompt surely?

File: BUILDING.md [hunk 2]
```diff
@@ -2,11 +2,13 @@
...
+
 First check out the git repo with `git clone`. Also make sure you get the submodules (`git submodule update --init` should be enough).
 
 There is one special external dependency, Boost:
 
-* On Windows, download it from <https://www.boost.org/users/download/> and extract it to `vendor/boost`.
+* On Windows, download it from <https://www.boost.org/users/download/> and extract the contents of the directory with a name like `boost_1_81_0` directory to `vendor/boost`.
```
Comment (hikari-no-yume) [inline BUILDING.md]: Oops, you wrote “directory” twice.

File: BUILDING.md (hunk 1 above)
Comment (nighto) [reply BUILDING.md]: It does not, it just does nothing. When you open it from a terminal, it displays several options. So I think it just outputs it to `STDOUT` and closes straight ahead.

image

File: BUILDING.md (hunk 2 above)
Comment (nighto) [reply BUILDING.md]: Oopsie, fixed

File: BUILDING.md (hunk 1 above)
Comment (hikari-no-yume) [reply BUILDING.md]: Wow… I feel weird about documenting this here though, I consider building to be a more advanced use-case so I expect the user will already be familiar with the command-line in that case. It seems like there's already an issue for it in the upstream repo: https://github.com/rust-lang/rustup/issues/2598

Would you be okay with dropping this line? I know it's helpful, but there's probably a lot of other things we'd need to add if we wanted to help users install a full toolchain.

Comment (nighto) [reply BUILDING.md]: Sure, no problem

**Pull request closed without merging**